SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp

```diff
@@ -492,22 +493,33 @@ AActor* USpatialReceiver::CreateActor(improbable::Position* Position, improbable
 
 void USpatialReceiver::ApplyComponentData(Worker_EntityId EntityId, Worker_ComponentData& Data, USpatialActorChannel* Channel)
 {
-	UClass* Class = TypebindingManager->FindClassByComponentId(Data.component_id);
-	checkf(Class, TEXT("Component %d isn't hand-written and not present in ComponentToClassMap."), Data.component_id);
+	uint32 Offset = 0;
+	bool bFoundOffset = TypebindingManager->FindOffsetByComponentId(Data.component_id, Offset);
+	if (!bFoundOffset)
+	{
+		return;
+	}
 
-	UObject* TargetObject = GetTargetObjectFromChannelAndClass(Channel, Class);
-	if (!TargetObject)
+	UObject* TargetObject = PackageMap->GetObjectFromUnrealObjectRef(FUnrealObjectRef(EntityId, Offset));
+	if (TargetObject == nullptr)
 	{
 		return;
 	}
-	FChannelObjectPair ChannelObjectPair(Channel, TargetObject);
 
-	FClassInfo* Info = TypebindingManager->FindClassInfoByClass(Class);
-	check(Info);
+	FClassInfo* Info = TypebindingManager->FindClassInfoByComponentId(Data.component_id);
```

Comment:
Nit: first `FindClassByComponentId`, then `FindClassInfoByClass`

Reply:
Since it can be a subobject, you want the class info of the specific component, not the Class (since that ClassInfo of a subobject won't have any associated components)